src/pages/reports/kitReports.js

```diff
@@ -0,0 +1,328 @@
+    const participantsActive = allParticipants.filter(
+        (element) => element["participation_status"] === "active"
+    );
+    const participantsWithdrawal = allParticipants.filter(
+        (element) => element["participation_status"] === "withdraw"
+    );
+
+    const participantsActiveSortDateAsc = participantsActive.sort((a, b) =>
+        a.time_stamp < b.time_stamp ? -1 : a.time_stamp > b.time_stamp ? 1 : 0
+    );
+    const participantsWithdrawalSortDateAsc = participantsWithdrawal.sort(
+        (a, b) => a.time_stamp < b.time_stamp ? -1 : a.time_stamp > b.time_stamp ? 1 : 0
+    );
+
+    // MERGE BOTH ARRAYS
+    return participantsActiveSortDateAsc.concat(
+        participantsWithdrawalSortDateAsc
+    );
+};
+
+const allParticipantsActive = (allParticipants) => {
+    const participantsActive = allParticipants.filter(
+        (element) => element["participation_status"] === "active"
+    );
+
+    const participantsActiveSortDateAsc = participantsActive.sort((a, b) =>
+        a.time_stamp < b.time_stamp ? -1 : a.time_stamp > b.time_stamp ? 1 : 0
+    );
+    return participantsActiveSortDateAsc;
+};
+
+/*
+==================================================
+UTIL FUNCTIONS 
+==================================================
+*/
+
+const createKitReportRows = (participantRows) => {
+    let template = ``;
+    participantRows.forEach((item) => {
+        template += `
+                        <tr class="row-color-enrollment-dark participantRow">
+                            <td style="text-align:center;">${daysBetween(
+                                item.time_stamp
+                            )}</td>
+                            <td style="text-align:center;">${convertTime(
+                                item.time_stamp
+                            )}</td>
+                            <td style="text-align:center;">${
+                                item.supply_kitId
+                            }</td>
+                            <td style="text-align:center;">${
+                                item.participation_status
+                            }</td>
+                        </tr>`;
+    });
+    return template;
+};
+
+const uniqueDaysElapsedObj = (allParticipantsActiveArr) => {
+    let daysElapsedArr = [];
+    allParticipantsActiveArr.forEach((item) => {
+        daysElapsedArr.push(daysBetween(item.time_stamp));
+    });
+    
+    const countObj = daysElapsedArr.reduce(
+        (acc, value) => ({
+            ...acc,
+            [value]: (acc[value] || 0) + 1,
+        }),
+        {}
+    );
+    return countObj;
+};
+
+// Calculate Number of days between current date and date provided
+const daysBetween = (
+    date1String,
+    date2String = new Date().toISOString().toLocaleString()
+) => {
+    const d1 = new Date(date1String);
+    const d2 = new Date(date2String);
+    return Math.floor((d2 - d1) / (1000 * 3600 * 24));
+};
+
+const convertTime = (time) => {
```

Comment:
Looks like you're using this function multiple locations. Can you move it to shared.js & only use the function call when required.

Reply:
Sounds good.